cmd/brew-bootstrap-jenv-java

```diff
@@ -0,0 +1,87 @@
+#!/bin/bash
+# Installs Java.
+set -e
+
+if [ "$1" = "--debug" ]; then
+  shift
+  PRINT_DEBUG="1"
+  set -x
+fi
+
+warn() { echo "$@" >&2; }
+abort() { EXPECTED_EXIT="1"; warn "$@"; exit 1; }
+
+abort_for_sh() {
+  abort 'Error: add `eval "$(jenv init -)"` to the end of your .bash_profile!'
+}
+
+abort_for_zsh() {
+  abort 'Error: add `eval "$(jenv init -)"` to the end of your .zshrc!'
+}
+
+abort_for_fish() {
+  abort 'Error: check the installation instructions at https://github.com/gcuisinier/jenv#gettings-started!'
+}
+
+abort_with_shell_setup_message() {
+  case $(basename ${SHELL:-bash}) in
+  sh|bash)
+    abort_for_sh
+    ;;
+  zsh)
+    abort_for_zsh
+    ;;
+  fish)
+    abort_for_fish
+    ;;
+  # tcsh users are on their own
+  *)
+    abort 'Error: you must finish setting up jenv in your shell; check `jenv init` for instructions!'
+  esac
+}
+
+cleanup() {
+  set +e
+  if [ -n "$EXPECTED_EXIT" ]; then
+    return
+  fi
+  warn "Error: $(basename $0) failed!"
+  if [ -z "$PRINT_DEBUG" ]; then
+    warn "For debugging output run:"
+    warn "  $0 --debug"
+    warn "If you're stuck: file an issue with debugging output at:"
+    warn "  https://github.com/github/homebrew-bootstrap/issues/new"
+  fi
+}
+trap "cleanup" EXIT
+
+BASE_PATH="$(cd "$(dirname "${BASH_SOURCE[0]}")/.." && pwd)"
+
+if ! which rbenv &>/dev/null; then
+  abort "Error: you need to 'brew install jenv'!"
+fi
+
+if ! jenv version-name &>/dev/null; then
```

Comment:
I'm not sure to understand the goal of this if clause 😕 

`jenv version-name` indicates the version of the currently configured JDK, following jEnv's priority semantics: if `.java-version` is present, then that's the version, otherwise fallback to the next in priority (No 💯 sure, it guess it would be the global JDK).

So if there is no JDK at all installed, `$JAVA_REQUESTED` won't be set, is that correct?
I think it would be OK to remove that wrapping if clause and just leave the ones inside, checking the env variable or a locally configured JDK

Reply:
`jenv version-name` exits 1 if the requested version isn't known to jenv; this is tracking that status rather than checking the value. We have the same logic in the rbenv and nodenv versions of the script.

Reply:
gotcha, so if it's known, wouldn't `$JAVA_REQUEST` not be defined, and if so, the script would fail in subsequent lines?

Reply:
`$JAVA_REQUESTED` is only assigned and used within the body of the `if` statement, which won't be reached if `jenv version-name` exits 1.

Reply:
duh, me dufus. Just realized the identation of the nested statements. Don't mind my comment 😓